scripts/scil_estimate_bundles_diameter.py

```diff
@@ -0,0 +1,348 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Script to estimate the diameter of bundle(s) along their length.
+The script expects:
+- bundles with coherent endpoints from scil_uniformize_streamlines_endpoints.py
+- labels maps with around 5-50 points scil_compute_bundle_voxel_label_map.py
+    <5 is not enough, high risk of bad fit
+    >50 is too much, high risk of bad fit
+- bundles that are close to a tube
+    without major fanning in a single axis
+    fanning is in 2 directions (uniform dispersion) good approximation
+
+The scripts print a JSON file with mean/std to be compatible with tractometry.
+WARNING: STD is in fact an ERROR measure from the fit and NOT an STD.
+
+Since the estimation and fit quality is not always intuitive for some bundles
+and the tube with varying diameter is not easy to color/visualize,
+the script comes with its own VTK rendering to allow exploration of the data.
+(optional).
+"""
+
+import argparse
+import json
+import os
+
+from dipy.io.utils import is_header_compatible
+from fury import window, actor
+import matplotlib.pyplot as plt
+import nibabel as nib
+import numpy as np
+from scipy.linalg import svd
+from scipy.ndimage import map_coordinates
+from scipy.ndimage.filters import gaussian_filter
+import vtk
+
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.io.utils import (add_overwrite_arg,
+                             add_reference_arg,
+                             add_json_args,
+                             assert_inputs_exist,
+                             parser_color_type)
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawTextHelpFormatter)
+    p.add_argument('in_bundles', nargs='+',
+                   help='List of tractography files supported by nibabel.')
+    p.add_argument('in_labels', nargs='+',
+                   help='List of labels maps that matches the bundles.')
+
+    p.add_argument('--fitting_func', choices=['lin_up', 'lin_down', 'exp',
+                                              'inv', 'log'],
```

Comment:
Please indicate what is the default option. Given our internal testing with Maggie and Matthieu, I would choose lin_up as default. 

Also, can you add a small documentation in something like here:
docs/source/documentation/bundle_diameter_fitting_function.rst

`[2021-12-13 12:31 p.m.] François Rheault
6 methods: (Les noms sont dans les filename), 2-6 sont dans l'ordre pour les CST)Default (pas montré pour le CST)lin_up: Moins de ponderation au stock proche du centre, plus vers l'extérieur (grossi enveloppe)lin_down: Plus de pondération au stock proche ( petite enveloppe)exp: Exponential (base 2.72) (entre 0 et 1),  pas mal plus de poids au stock proche (petite enveloppe)inv: 1 / X, BEAUCOUP plus de poids au stock proche (mini enveloppe)log: log naturel, peu de poids au centre, stable a une certaine distance (grossi l'enveloppe)

(Les functions sont appliqués sur les distances au centre, donc c'est pas des thresholds ou des parametrers défini par le user. Le user pick une fonction.La fonction inverse, c'est surement juste bon pour détecter des stems Laurent style)`

Reply:
`[2021-12-13 12:40 p.m.] François Rheault
Faut faire attention, c'est yet another degree of freedom. On peut pas commencer à utiliser une fonction pour tel bundle et une autre pour l'autre bundle. Sinon ça ferait pas de sens pour votre métrique de densité
`